framework/DataObjects/DataSet.py

```diff
@@ -1218,6 +1218,12 @@ def _convertToDict(self):
     # supporting data
     dataDict['dims']     = self.getDimensions()
     dataDict['metadata'] = self.getMeta(general=True)
+    dataDict['type'] = self.type
+    dataDict['inpVars'] = self.getVars('input')
+    dataDict['outVars'] = self.getVars('output')
+    dataDict['numberRealization'] = self.size
+    dataDict['name'] = self.name
+    dataDict['metaKeys'] = self.getVars('meta')
```

Comment:
I am conflicted here (I mean for the usage of this for PP). I do not think we should use dictionaries anymore in PP). The data should be provided as Dataset only. What do you think?

Reply:
I wonder if we can provide sufficiently clear examples for developers so that Datasets are easy for them to work with for the things they need to do; otherwise, I worry this could be a big barrier to external development, since the xarray datasets can be difficult until you've played with them a bit, at least for me.

Reply:
I leave the option to use either Dict or Dataset. Currently, we only have BasicStatistics to use Dataset. With this option, we can identify the PP that we need to convert to use Dataset, and convert them one by one. Otherwise, we can not merge the  PP changes without converting all PPs to use Dataset. In addition, I'm not sure it is worth to convert our PPs to use Dataset if there is no need from our customer.